app/index/activities/route.js

```diff
@@ -101,9 +101,20 @@ export default Route.extend({
      * @public
      */
     generateReportsCheck() {
-      this.get('controller.activities').filterBy('task.id', undefined).length
-        ? this.set('controller.showWarning', true)
-        : this.send('generateReports')
+      let hasUnknown = this.get('controller.activities').find(a => {
+        return !a.get('task.id')
+      })
+
+      let hasOverlapping = this.get('controller.activities').find(a => {
+        return a.get('overlaps')
+      })
```

Comment:
How about `!!this.get('controller.activities').findBy('overlaps')`
and 
`!!this.get('controller.activities').findBy('task.id', false)`